[marcus-ny] iP

marcus-ny opened the pull request:
## Solaire

> Use this, to summon one another as spirits, cross the gaps between the worlds, and engage in jolly co-operation! - :sunny: Solaire of Astora, Dark Souls

Solaire of Astora will help you remember your daily tasks. It's:
- text-based
- easy to learn
- ~~not so fast~~ fast

All you need to do is:
1. Download from [here](https://github.com/marcus-ny/ip/releases/tag/A-Jar)
2. Run it
3. Watch it do your chores

And it's **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines
- [ ] Reminders (coming soon)

If you're interested to find out more about the implementaiton, start from the `Solaire` class.

```
public static void main(String[] args) {
        Solaire solaire = new Solaire();
        solaire.startConversation();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

marcus-ny committed:
feature: implement level-1

- Add simple greeting and farewell message

marcus-ny committed:
refactor: move chatbot to own class

marcus-ny committed:
implement level-1

- Add echo behaviour to Solaire

marcus-ny committed:
implement level-2

- added a To-do-list that stores list of user inputs and print them on request

marcus-ny committed:
implement level 3

marcus-ny committed:
add different sub-task categories

- add Deadline, Event and Todo classes that extend Task class

marcus-ny committed:
implement level 4

- add different commands for different task types

marcus-ny committed:
implement testing scripts

- write sample test inputs and outputs

marcus-ny committed:
implement chatbot error

- implement new exception called SolaireException
- handle incorrect input text formats and recommend the correct format to user

marcus-ny committed:
implement task delete

marcus-ny committed:
Format Solaire.java

marcus-ny committed:
Add TaskIO.java

Reads local file input and converts to relevant Tasks

marcus-ny committed:
Add file writing capabilities

Solaire can now write and read tasks from a txt file

marcus-ny committed:
Merge pull request #1 from marcus-ny/branch-Level-7

Branch level 7

marcus-ny committed:
Modify Deadline.java's deadline datatype

Modify deadline's datatype from String to LocalDate to support formatted date and time

marcus-ny committed:
Alter deadline format name

marcus-ny committed:
Alter variable name of formatted date

marcus-ny committed:
Merge branch 'branch-Level-8'

marcus-ny committed:
Create TaskList.java

marcus-ny committed:
Create Parser.java

marcus-ny committed:
Update TaskIo class name to Storage

marcus-ny committed:
Refactor Solaire.java

Connect TaskList, Ui, Parser and Storage classes to create a more OO structure in Solaire

marcus-ny committed:
Merge branch-A-MoreOOP into master

marcus-ny committed:
Organize files into packages

marcus-ny committed:
Merge branch-A-Packages into master

marcus-ny committed:
Merge add-gradle-support to master

marcus-ny committed:
Move files to accommodate Gradle build functions

marcus-ny committed:
Create JUnit testcases for Deadline and TaskList

marcus-ny committed:
Merge branch-A-JUnit to master

marcus-ny committed:
Storage: handle cases where file is missing

marcus-ny committed:
Storage: handle missing directory case on write

marcus-ny committed:
Merge branch-fix-Level-7 to master

marcus-ny committed:
Update gradle build name

marcus-ny committed:
Storage: fix file directory creation error

Fix a bug where there is an IOException on creating the target txt file when it does not exist

marcus-ny committed:
Merge branch-A-Jar to master

marcus-ny committed:
Create JavaDoc for relevant public classes and methods

marcus-ny committed:
Modify code to comply with coding standards

marcus-ny committed:
Add Find function to Solaire

The Find command matches the existing tasks' descriptions with the given prompt and filters the list accordingly

marcus-ny committed:
Merge branch-A-JavaDoc to master

marcus-ny committed:
Merge branch-A-CodingStandard to master

marcus-ny committed:
Merge branch-Level-9 to master

marcus-ny committed:
Fix regression test compilation errors

marcus-ny committed:
Fix checkstyle errors

marcus-ny committed:
Merge branch 'branch-A-Checkstyle'

javierng2knus commented on src/main/java/solaire/data/task/Task.java:
```diff
@@ -0,0 +1,54 @@
+    private static int count = 0;
+    private int id;
+    private String taskName;
+    private Boolean isDone;
```
Great naming of boolean members and methods. 👍

marcus-ny committed:
implement GUI

- modify print statements to String return type to send to GUI components

marcus-ny committed:
Change user and chatbot images

marcus-ny committed:
Change background color of chatbox

marcus-ny committed:
Merge branch-Level-10 into master

marcus-ny committed:
Add GUI app build options

marcus-ny committed:
Merge branch-Level-10 into master

javierng2knus commented on src/main/java/solaire/tasklist/TaskList.java:
```diff
@@ -0,0 +1,143 @@
+        for (Task item : taskList) {
+            if (item.getId() == id) {
+                item.markAsDone();
+                System.out.print("Marked item number: " + item.getId() + "\n");
```
Should the functionality of the printing of UI in the TaskList class be extracted so that it returns a string representation of the list of tasks, which the UI class can then handle?

javierng2knus commented on src/main/java/solaire/parser/Parser.java:
```diff
@@ -0,0 +1,61 @@
+     */
+    public static Task parseTaskInput(String input) throws SolaireException {
+        if (input.startsWith("deadline")) {
+            Matcher deadlinePattern = Pattern.compile("^(?i)deadline\\s+(.+)\\s+/by\\s+(\\S+.*)").matcher(input);
```
I like your usage of the Matcher class as a means to search for the specific patterns in the user inputs! 👍

javierng2knus reviewed (commented):
Fantastic work!
Very exhaustive JavaDoc and great adherence to the coding standards.
LGTM! 👍

marcus-ny committed:
Add assertions to perform checks for invalid inputs

There are no checks to assure that certain user inputs are valid

There are methods where user inputs are parsed but no check is performed to ensure non-empty.

Let's use assertions to ensure that inputs are not empty before parsing.

marcus-ny committed:
Refactor Parser.java

Method `parseTaskInput` in `Parser` is long and exceeds 30 LOC

Let's extract the parsing of individual task type to corresponding functions for better readibility

marcus-ny committed:
Refactor Storage.java

The method `parseTasks` in `Storage` is long and exceeds 30 LOC.

Let's refactor by extracting the task creation to a separate method to make the code more readable.

marcus-ny committed:
Refactor Storage

File path for the chatbot's data is hard-coded.

This is less safe since it is a magic literal. Changing the file path would require many repeated changes.

Moreover, the file path should be a constant.

Let us extract the `filePath` variable to be a static variable associated with `Storage` class.

marcus-ny committed:
Merge pull request #2 from marcus-ny/branch-A-Assertions

Increment A-Assertions

marcus-ny committed:
Merge branch 'master' into branch-A-CodeQuality

marcus-ny committed:
Merge pull request #3 from marcus-ny/branch-A-CodeQuality

Increment A-CodeQuality

marcus-ny committed:
Add CI workflow

marcus-ny committed:
Merge branch-A-CI into master

marcus-ny committed:
Disable ./runtest.sh checks

The app no longer operates on CLI and hence text-ui-tests are no longer necessary

marcus-ny committed:
Fix missing file writing capabilities

marcus-ny committed:
Merge branch-fix-localDiskAccess to master

marcus-ny committed:
Add reminders for deadline tasks

Users have no way to easily see which tasks are due within a certain time frame.

Let's add a reminder feature called by command `remindme` so that users can see a list of tasks due within specified time frame.

marcus-ny committed:
Merge branch-B-Reminders into master

youdonnnn commented on src/main/java/solaire/Solaire.java:
```diff
@@ -0,0 +1,86 @@
+    private TaskList taskList;
+    private Ui ui;
+
+    enum UserCommands {
```
Maybe you might want to create a separate enums class UserCommands instead of what is done here? 🤔

youdonnnn commented on src/main/java/solaire/Solaire.java:
```diff
@@ -0,0 +1,86 @@
+        assert inputCommand.length > 0 : "Input command should not be empty";
+        try {
+            UserCommands command = UserCommands.valueOf(inputCommand[0].toUpperCase());
+            switch (command) {
```
LGTM! I liked the way you handled the different input commands with a switch 👍

youdonnnn commented on src/main/java/solaire/data/task/Deadline.java:
```diff
@@ -0,0 +1,63 @@
+    }
+
+    @Override
+    public String toString() {
```
It might be easier to use a formatter to convert the date to a format that you want and then print it out, it would reduce the need for you to extract the month, day and year individually here!

youdonnnn commented on src/main/java/solaire/data/task/Deadline.java:
```diff
@@ -0,0 +1,63 @@
+package solaire.data.task;
+
+import java.time.LocalDate;
```
Maybe using LocalDateTime would be better? 🤔

youdonnnn commented on src/main/java/solaire/ui/Ui.java:
```diff
@@ -0,0 +1,51 @@
+                        + "father!\n" + "If only I could be so grossly incandescent!\n";
+
+        return greetingMessage;
+        //lineBreak();
```
LGTM! Although it might be better to remove this commented function if it is not going to be used!

marcus-ny committed:
Add greeting message on launch

marcus-ny committed:
Fix cut off dialogbox in java GUI app

marcus-ny committed:
Remove commented code

marcus-ny committed:
Update user image to square orientation

marcus-ny committed:
Implement GUI exit on "bye" command

marcus-ny committed:
Introduce delay before exitting GUI

marcus-ny committed:
Merge branch-A-BetterGui into master

marcus-ny committed:
Remove unused import in DialogBox.java

marcus-ny committed:
Remove unused method linebreak() in TaskList.java

marcus-ny committed:
Fix JavaDoc

marcus-ny committed:
Set title for GUI

marcus-ny committed:
Fix local disk reading bug for Event tasks

marcus-ny committed:
Add Ui.png

marcus-ny committed:
Add README.md

marcus-ny committed:
Merge branch-finalChecks into master

marcus-ny committed:
Update README.md for repository home page

marcus-ny committed:
Fix negative reminder buffer

marcus-ny committed:
Merge branch-fixReminder into master
